src/bitvec.nim

```diff
@@ -0,0 +1,56 @@
+# bitvec
+# Copyright KeyMe
+# Extensible Bit Vector library for nim
+import options
+import sequtils
+
+proc pop[T](s: var seq[T], idx: int): T =
+  result = s[idx]
+  s.delete(idx)
+
+proc encode*[T: SomeInteger](n: T): seq[byte] =
+  ## Returns a list containing the encoded integer
+  result = @[]
+
+  if n == 0:
+    return @[0.byte]
+
+  var num = n
+  while num > 0.T:
+    var v = num mod 128
+    num = num div 128
+    if num > 0.T:
+      v = v or 0x80
+    result.add(v.byte)
+
+proc decode*[T: SomeInteger](input: openarray[byte]): (Option[T], seq[byte]) =
+  ## Returns a tuple containing the first decoded integer and
+  ## the remainder of the input list
+
+  if len(input) == 0:
+    return
+
+  # Duplicate the input as a mutable sequence
+  var ls = @input
+  var digit = ls.pop(0)
+  var resVal = 0.T
+
+  #If this is larger than a single byte, build up the integer
+  var mul = 1
+  while true:
```

Comment:
Just a code suggestion but if you have the while len(ls) > 0 and then move the pop from above and put is as the first line you won't need the bottom if else.